build/pipelines/Build.jenkinsfile

```diff
@@ -0,0 +1,120 @@
+    }
+
+    stages
+    {
+        stage('Checkout')
+		{
+			steps
+			{
+				deleteDir()
+				checkout(
+                    changelog: true,
+                    poll: true,
+                    scm: [
+                        $class: 'GitSCM',
+                        branches: scm.branches,
+                        extensions: scm.extensions + 
+                            [[
+                                $class: 'CloneOption',
+                                shallow: true,
+                                noTags: true,
+                                reference: '',
+                                timeout: 10
+                            ]],
+                        userRemoteConfigs: scm.userRemoteConfigs
+                    ]
+                )
+			}
+		}
+        stage('Read properties')
+        {
+            steps
+            {
+                script
+                {
+                    println 'Reading pipeline properties'
+                    props = readProperties file:'build/pipelines/pipeline.properties'
+                    props.each { propKey, propValue -> println "${propKey}: ${propValue}" }
+                }
+            }
+        }
+        stage('Build')
+        {
+            steps
+            {
+                script
+				{
+					props.projectConfigurations.split(',').each
+					{ config ->
+						props.buildProjects.split(',').each
+						{ project ->
+								sh script: "build/scripts/build.sh -p '${project}' -c '${config}' -r '$WORKSPACE' -s '${props.solutionPath}' -d '$VS150COMNTOOLS\\..\\IDE\\devenv.com' -v " + "${version}" + " -n " + "${channel}", label: "Build project '${project}' for '${config}'"
+						}
+					}
+				}           
+            }
+        }
+        stage('Static Code Analysis')
+		{
+			steps
+			{
+				sh "build/scripts/cppcheck_execute.sh -r '$WORKSPACE' -p '${props.cppCheckProjectPaths}' -k '${props.cppCheckProjectNames}'"
+			}			
+		}
+
+        stage('SonarQube Upload')
+		{
+			steps
+			{
+				script
+				{
+					try
+					{
+						scannerHome = tool 'SonarQubeScanner'
+						withSonarQubeEnv('SonarQube')
+						{
+							bat script: "${scannerHome}bin\\sonar-scanner.bat -Dproject.settings=sonar-project.properties -Dsonar.branch.name=${env.BRANCH_NAME} -Dsonar.projectVersion=${version}/${channel}", label: "SonarQube analysis"
+						}
+						timeout(time: 15, unit: 'MINUTES')
+						{
+							waitForQualityGate abortPipeline: false
+						}
+					}
+					catch (Exception err)
+					{
+						currentBuild.result = 'UNSTABLE'
+					}
+				}
+			}
+		}
+    }
+    post
+    {
+        always
+        {
+            // cobertura autoUpdateHealth: false, autoUpdateStability: false, coberturaReportFile: "${props.codeCoverageReportFiles}", failNoReports: false, failUnhealthy: false, failUnstable: false, lineCoverageTargets: '30, 0, 0', maxNumberOfBuilds: 0, onlyStable: false, sourceEncoding: 'ASCII', zoomCoverageChart: false
```

Comment:
No gtest project after all and I'm not sure with the approach we are aiming for with javascript we would be able to get any coverage out of it. If not, I suggest removing this line and associated info on pipeline.properties.

Reply:
I have left it here until the test is done and we are sure of that.